pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -495,4 +499,104 @@ public IdealState apply(@Nullable IdealState idealState) {
       String tenant) {
     return new HashSet<>(getInstancesConfigsWithTag(instanceConfigs, TagNameUtils.getBrokerTagForTenant(tenant)));
   }
+
+  /**
+   * Add default tags to instance if the instance has no tags Note for the getDefaultTags: it is a
+   * lambda so it may not be invoked if instance already has tags. * For example () ->
+   * ImmutableList.of("Default_tenant")
+   *
+   * @param helixManager The helix manager for update
+   * @param instanceId the Helix instance Id
+   * @param getDefaultTags the default tags lambda. Something like () -> ImmutableList.of("Default_tenant") to provide default tags
+   */
+  public static void addDefaultTags(
+      HelixManager helixManager, String instanceId, Supplier<List<String>> getDefaultTags) {
```

Comment:
Pass in an `InstanceConfig` and a tag supplier, and return a boolean denoting whether the config is changed. We don't want to read an instance config and post it back for every update. We should only read the instance config once, track whether it is updated, and only post it back once if there are changes.

Reply:
All updated.
Now the tracking logic is also in HelixHelper so the code for 4 host types are shared.